### PR TITLE
Implements Feature : Send email to users to notify their daily transactions at 00:00

### PR DESCRIPTION
**Pull request by Hamza-Chaudhary65**
Uses `Celery` with `redis` to send email asynchronously.

### CODE REVIEW

**Review by ZAINABFATIMA0 — APPROVED** [review]

File: fintrack/fintrack/settings.py [hunk 1]
```diff
@@ -144,3 +146,22 @@
         'rest_framework.permissions.IsAuthenticated',
     ],
 }
+
+
+EMAIL_BACKEND = 'django.core.mail.backends.smtp.EmailBackend'
+EMAIL_HOST = config('EMAIL_HOST')
```
Comment (sheralim012) [inline fintrack/fintrack/settings.py:151]: update .env.example file

Comment (sheralim012) [reply fintrack/fintrack/settings.py:151]: create readme for this

File: fintrack/fintrack/settings.py [hunk 2]
```diff
@@ -144,3 +146,22 @@
...
+EMAIL_PORT = config('EMAIL_PORT')
+EMAIL_USE_TLS = config('EMAIL_USE_TLS')
+EMAIL_HOST_USER = config('EMAIL_HOST_USER')
+EMAIL_HOST_PASSWORD = config('EMAIL_HOST_PASSWORD')
+
+CELERY_BROKER_URL = 'redis://localhost:6379/0'
+
+CELERY_BEAT_SCHEDULER = 'django_celery_beat.schedulers:DatabaseScheduler'
+CELERY_TIMEZONE = 'Asia/Karachi'
```
Comment (sheralim012) [inline fintrack/fintrack/settings.py]: why change this?

File: fintrack/fintrack/settings.py [hunk 3]
```diff
@@ -144,3 +146,22 @@
...
+CELERY_BEAT_SCHEDULE = {
+    'send_transaction_emails': {
+        'task': 'transactions.tasks.send_transaction_emails',
```
Comment (sheralim012) [inline fintrack/fintrack/settings.py]: I would suggest to use django admin for scheduling task, that way upon change, deployment wouldn't have to done

File: fintrack/transactions/tasks.py [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+@shared_task
+def send_transaction_emails():         
+    transactions = Transaction.objects.filter(
+        date=timezone.now().date()
+    )
+    user_emails = transactions.values_list(
+        'transaction_from_account__holder__email',
+        flat=True
+    ).distinct()
```
Comment (sheralim012) [inline fintrack/transactions/tasks.py]: wouldn't it be much easier to loop through users and then fetch all of the transaction osf the current user?

File: fintrack/transactions/tasks.py [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+    messages = []
+
+    for user_email in user_emails:
+        user_transactions = transactions.filter(
+            transaction_from_account__holder__email=user_email
```
Comment (sheralim012) [inline fintrack/transactions/tasks.py]: it's always better to user foreign key to fetch related objects or related name to do that

File: fintrack/transactions/tasks.py [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+        )
+        message = (
+            "Daily Transaction Report",
+            transaction_list,
+            config('EMAIL_HOST_USER'),
+            [user_email]
+        )
+        messages.append(message)
+    
+    mail.send_mass_mail(messages, fail_silently= False)
```
Comment (sheralim012) [inline fintrack/transactions/tasks.py]:
```suggestion
    mail.send_mass_mail(messages, fail_silently=False)
```

File: fintrack/transactions/tasks.py [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+        transaction_list = "\n".join(
+            [
+                f" Category : {transaction.category},"
+                f" Amount : {transaction.amount}, "
+                f" Description : {transaction.description},"
+                f" Account {transaction.transaction_from_account}" for transaction in user_transactions
+            ]
+        )
+        message = (
+            "Daily Transaction Report",
```
Comment (sheralim012) [inline fintrack/transactions/tasks.py]: instead of compiling email body like this, read about HTML body using template

**Review by sheralim012 — CHANGES_REQUESTED** [review]

File: README.md [hunk 8]
```diff
@@ -1 +1,70 @@
...
+CELERY_BROKER_URL = 'redis://localhost:6379/0'
+CELERY_RESULT_BACKEND = 'redis://localhost:6379/0'
+
+
+2. Install a Message Broker
+Celery needs a message broker to send and receive messages. Install Redis with pip:
+
+pip install redis
+
+3. Configure Celery in Your Project
```
Comment (sheralim012) [inline README.md]: This section isn't necessary as it's part of the codebase

File: fintrack/fintrack/settings.py [hunk 9]
```diff
@@ -117,7 +119,7 @@
 
 LANGUAGE_CODE = 'en-us'
 
-TIME_ZONE = 'UTC'
+TIME_ZONE = 'Asia/Karachi'
```
Comment (sheralim012) [inline fintrack/fintrack/settings.py]: Again why are we changing default also it's recommended to store everything in UTC and convert according to necessary

**Review by sheralim012 — CHANGES_REQUESTED** [review]

File: README.md (hunk 8 above)
Comment (sheralim012) [reply README.md]: this isn't addressed

**Review by sheralim012 — CHANGES_REQUESTED** [review]

**Review by sheralim012 — APPROVED** [review]

**Pull request merged**